js/taskOverlay/taskOverlay.js

```diff
@@ -216,7 +218,60 @@ var taskOverlay = {
     }
   },
 
+  initializeSearch: function () {
+    var container = document.querySelector('.task-search-input-container')
+    var input = document.getElementById('task-search-input')
+
+    container.addEventListener('click', e => { e.stopPropagation(); input.focus() })
+
+    input.addEventListener('input', function (e) {
+      var search = input.value.toLowerCase().trim()
+
+      if (!search) {
+        // reset the overlay
+        taskOverlay.show()
+        input.focus()
+        return
+      }
+
+      tasks.forEach(function (task) {
+        var taskContainer = document.querySelector(`.task-container[data-task="${task.id}"]`)
+
+        var tabMatches = 0
+        task.tabs.forEach(function (tab) {
+          var tabContainer = document.querySelector(`.task-tab-item[data-tab="${tab.id}"]`)
+
+          var searchText = (tab.title + ' ' + tab.url).toLowerCase()
```

Comment:
There could be also a search by task name:
```js
          const taskName = taskContainer.querySelector('input.task-name').getAttribute('placeholder')
          var tabContainer = document.querySelector(`.task-tab-item[data-tab="${tab.id}"]`)

          var searchText = (`${taskName} ${tab.title} ${tab.url}`).toLowerCase()
```
What do you think?

Reply:
Yeah good point, I did something like this in cbddc1fcaf6abb6feefef458593c3c10d855cf56. Compared to your solution, it won't match the default task names (like "task 1"), but I think it's unlikely that anyone would search for that.

I also changed it so that for a multi-word search, the words are allowed to be in any order within the tab / task name.